guides/hack/05-functions/30-function-references.md

```diff
@@ -0,0 +1,183 @@
+$fun = Baz\derp<>;
+```
+
+```
+use function Foo\Bar\Baz\derp;
+$fun = derp<>;
+```
+
+## Static methods
+
+Similarly, you can refer to a static method `bar` on a class `MyClass` by using
+the familiar method call syntax, without providing the call arguments. Just
+append type arguments (like `<>`) to the function call receiver
+(like `MyClass::bar`).
+
+```
+MyClass::bar<>; // a reference to static method 'MyClass::bar'
+```
+
+- **Private/protected** methods can be referenced using this syntax as long as
+  they are accessible to you in your local scope.
+- **Abstract** static methods cannot be referenced. Such methods cannot be
+  called, for they have no implementation. Invoking a hypothetical reference to
+  one would also be an error, so we simply don’t allow a reference to be
+  created.
+- In traits and other classes that are not marked `final`, you cannot use
+  `self::` in a reference. This is to avoid ambiguity and confusion around what
+  `self` actually refers to when the method is called (it depends on the static
+  context of a call, whereas function references can never receive the context
+  information because the target is only resolved once).
+- Also, `parent::` is never supported.
+
+## Generics
+
+If you wish to pass along explicit generic arguments, either as a hint to the
+type checker, or in the case of a function with reified type parameters when
+they are required, that is also supported:
+
+```
+i_am_erased<int, _>; // erased generics (note wildcard)
+i_am_reified<C<string>>; // a reified generic
+```
+
+- Keep in mind that generics, if any, still must be provided at the location
+  where the function reference is created, rather than where it is used/invoked.
+- The arity of the type argument list, if it is non-empty (i.e. not `<>`), must
+  match the declaration, just like for an ordinary function call.
+- The special wildcard specifier `_` may be provided in place of any or all
+  erased (non-reified) generic arguments if you want Hack to infer a type
+  automatically based on the type parameter’s constraints. Again this is the
+  same as for ordinary function calls.
+
+Example with erased generics:
+
+```
+function fizz<Ta as num, Tb>(Ta $a, Tb $b): mixed { ... }
+
+$x = fizz<int, string>; // OK
+$x(4, 'hello');
+$x(-1, false); // error!
+
+$y = fizz<>;
+$y(3.14, new C()); // also OK
+$y('yo', derp()); // error!
+
+// OK as well
+$z = fizz<int, _>;
+$z = fizz<_, string>;
+$z = fizz<_, _>;
+
+// these all have errors!
+fizz<_>;
+fizz<string, _>;
+fizz<string, int>;
+```
+
+Example with [reified generics](/hack/generics/reified-generics):
+
+```
+function buzz<reify T as arraykey>(T $x): mixed { ... }
+
+$w = buzz<int>; // OK
+$w(42);
+$w("goodbye"); // error!
+
+// these all have errors!
+buzz<>;
+buzz<_>;
+buzz<mixed>;
+```
+
+## Introspection
+
+Function references can be cached in APC (the name will be resolved again when
+they are rehydrated) or passed to memoized functions. However, other
```

Comment:
rehydrated is an unusual term - maybe 'unserialized', 'retrieved from APC' or similar?